quickwit/quickwit-indexing/src/actors/indexing_service.rs

```diff
@@ -472,65 +483,69 @@ impl IndexingService {
         ctx: &ActorContext<Self>,
         physical_indexing_plan_request: ApplyIndexingPlanRequest,
     ) -> Result<ApplyIndexingPlanResponse, IndexingError> {
-        let mut updated_pipeline_ids: HashSet<IndexingPipelineId> = HashSet::new();
-        let mut pipeline_ordinals: HashMap<&IndexingTask, usize> = HashMap::new();
-        for indexing_task in &physical_indexing_plan_request.indexing_tasks {
-            let pipeline_ord = pipeline_ordinals.entry(indexing_task).or_insert(0);
-            let pipeline_id = IndexingPipelineId {
-                node_id: self.node_id.clone(),
-                index_uid: IndexUid::from(indexing_task.index_uid.to_string()),
-                source_id: indexing_task.source_id.clone(),
-                pipeline_ord: *pipeline_ord,
-            };
-            *pipeline_ord += 1;
-            updated_pipeline_ids.insert(pipeline_id);
-        }
-
-        let running_pipeline_ids: HashSet<IndexingPipelineId> =
-            self.indexing_pipelines.keys().cloned().collect();
+        let pipelines_uid_in_plan: FnvHashSet<PipelineUid> = physical_indexing_plan_request
+            .indexing_tasks
+            .iter()
+            .flat_map(|indexing_task| indexing_task.pipeline_uid.into_iter())
```

Comment:
Wherever we flatmap on the `pipeline_uid` option, we silently skip the empty, which in practice should not happen.

I wonder if instead we want to add a `pipeline_uid()` accessor that `unwrap`s or `unwrap_or_default`. The former will let us know right away if something goes wrong at the risk of panicking. The latter will be also silent but will at least appear in the logs. WDYT?